### PR TITLE
Module 3 Bookdtore Setup

### PR DESCRIPTION
**Pull request by the-catalystmc**
In this project I: 
Make sure that there are no linter errors.
Make sure that I use the correct flow of Gitflow.
Make sure that I documented your work in a professional way.
Initialize React app.
Add React-Redux (npm install react-redux).
Structure my application files using a "feature folder" approach and use the ducks pattern for my Redux files.
Add React Router and set two <Route>s and <Link>s for the app's navigation:
Books
Default view - will display the list of books and add/remove book functionality.
Categories
Will display "Under construction" text only.
Styling is not required at this point.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById('root'),
```
Comment (arslanbisharat) [inline src/index.js:9]: - [x] Please make sure the index has 2 `<Link>`s in the navbar which leads to the booklist and categories locations.
The navbar might be in its own component, that's ok!


- [x] Please make sure the index has 2 `<Route>`s, one for the BookList component and the other for the Categories component. In order to not show a blank page when visiting the site for the first time, the BookList component should be the default page

 


I can see you have implemented this in App.j please move that here for now.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] All the unnecessary files should be removed to make your code clean.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js (hunk 1 above)
Comment (the-catalystmc) [reply src/index.js:9]: @arslanbisharat I was not required to include the LInks or Routes in the index.js. Could you please enlighten me as to the purpose of moving them "for now"?

**Comment by arslanbisharat** [comment]
This is one of the requirements for now. Please make the changes and
request another review

**Comment by the-catalystmc** [comment]
@arslanbisharat If you try the site, you will see that everything works as required, my Routes and LInks are passed directly into the index.js through their components. I do not have a blank page and the booklist page is indeed the default page the Links inherits from the Navbar.js into the App.js and then into index.js. You are honestly confusing me because all my teammates got approved doing it like this and there is no explicit requirement like the one you are giving me. How does what I am doing go against project protocols?

**Comment by arslanbisharat** [comment]
> @arslanbisharat If you try the site, you will see that everything works as required, my Routes and LInks are passed directly into the index.js through their components. I do not have a blank page and the booklist page is indeed the default page the Links inherits from the Navbar.js into the App.js and then into index.js. You are honestly confusing me because all my teammates got approved doing it like this and there is no explicit requirement like the one you are giving me. How does what I am doing go against project protocols?
> 

I am following the guidelines. And please re-read my comment where I mentioned I can see you have added that feature but please move that to index.js

**Review by arslanbisharat — APPROVED** [review]
#
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**